seacrowd/sea_datasets/tgl_profanity/tgl_profanity.py

```diff
@@ -0,0 +1,115 @@
+import csv
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+from datasets.download.download_manager import DownloadManager
+
+from seacrowd.utils import schemas
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Licenses, Tasks
+
+_CITATION = """
+@article{galinato-etal-2023-context,
+    title="Context-Based Profanity Detection and Censorship Using Bidirectional Encoder Representations from Transformers",
+    author="Galinato, Valfrid and Amores, Lawrence and Magsino, Gino Ben and Sumawang, David Rafael",
+    month="jan",
+    year="2023"
+    url="https://papers.ssrn.com/sol3/papers.cfm?abstract_id=4341604"
+}
+"""
+
+_LOCAL = False
+_LANGUAGES = ["tgl"]
+_DATASETNAME = "tgl_profanity"
+_DESCRIPTION = """\
+This dataset contains 13.8k Tagalog sentences containing profane words, together
+with binary labels denoting whether or not the sentence conveys profanity /
+abuse / hate speech. The data was scraped from Twitter using a Python library
+called SNScrape and annotated manually by a panel of native Filipino speakers.
+"""
+
+_HOMEPAGE = "https://huggingface.co/datasets/mginoben/tagalog-profanity-dataset/"
+_LICENSE = Licenses.UNKNOWN.value
+_SUPPORTED_TASKS = [Tasks.SENTIMENT_ANALYSIS]
```

Comment:
I think the task is not exactly sentiment_analysis. Would it be possible to find another task name for this? Probably, ABUSIVE_LANGUAGE_PREDICTION ? @sabilmakbar @holylovenia any suggestions?

Reply:
A couple of points from me:
1. I was thinking that the use of profane language is similar to negative sentiment; hence, the task is still relevant but needs proper label naming in the definition.
2. regarding the task definition itself, if we decide to make a new task, will it be distinguishable between sentiment analysis w/ labels of negative & non-negative vs. abusive language prediction? If the line can be drawn clearly and the benefits outweigh more, we can proceed w/ the idea of creating a new task type of abusive language identification.

Reply:
prob we need @holylovenia or @SamuelCahyawijaya take on this

Reply:
btw, we might have another dataloader that can use this new task if we decide to create it.
https://github.com/SEACrowd/seacrowd-datahub/pull/284

Reply:
I agree with @gentaiscool's suggestion. Although the current dataset supports a binary label as @sabilmakbar has pointed out, using a new task would be better for the separation of concerns.

Could you please add the `ABUSIVE_LANGUAGE_PREDICTION` task in a separate PR and adjust the current dataloader for this task, @ljvmiranda921?

cc: @sabilmakbar @SamuelCahyawijaya

Reply:
I'll add this as I have another task add pending, @holylovenia. Will create the PR quickly

Reply:
@sabilmakbar Thanks Sabil! I've approved and merged #299.

cc: @ljvmiranda921

Reply:
Thanks @sabilmakbar for doing the PR! I updated this PR to reflect the new task.